### PR TITLE
(#384) Fix puppetdb request when the fact is a boolean

### LINKED ISSUE #384: choria puppetdb discovery doesn't work with boolean facts
Hello,

The choria discovery method with puppetdb can't discover hosts when using boolean facts. For example : 

    % mco find -F "some_fact==true" --dm=choria

Indeed, the puppetdb request made by choria add quotes around fact values which doesn't work when the fact is a boolean.

### PR DESCRIPTION
**Pull request by snoir**
Fix #384

### CODE REVIEW

**Comment by ripienaar** [comment]
Won’t this break non Boolean true/false?

We could look for true or “true” and keep current behaviours while also fixing the shortcoming?

**Comment by snoir** [comment]
You mean test if the fact value used for discovery is between quotes (or not) ? Like this : 

    % mco find -F "some_fact=='true'" --dm=choria

**Comment by ripienaar** [comment]
Nah. If someone filters for foo=true then we check in PuppetDB for `true` but also for `”true”` in the PQL query that way we catch all cases

**Comment by snoir** [comment]
I see, like this : 

    (inventory {facts.is_virtual = true or facts.is_virtual = "true"})

**Comment by ripienaar** [comment]
@snoir yeah - if thats valid PQL been a while since I used it hehe

**Comment by ripienaar** [comment]
looks good, if you can fix those rubocop failures this is good to go!

File: lib/mcollective/discovery/choria.rb [hunk 1]
```diff
@@ -132,9 +132,17 @@ def discover_facts(filter)
 
             'inventory {facts.%s ~ "%s"}' % [fact, regex]
           when "=="
-            'inventory {facts.%s = "%s"}' % [fact, value]
+            if value == "true" || value == "false"
```
Comment (ripienaar) [inline lib/mcollective/discovery/choria.rb]: these should be `["true", true].include?(value)` to shut up rubocop, ditto the other one

**Comment by snoir** [comment]
Indeed this is better :)
If it's ok I can regroup in a single commit

**Comment by ripienaar** [comment]
yeah looking good, squash to a single commit and we're good to go, make the commit message first line `(#384) Fix puppetdb request when the fact is a boolean`

**Comment by ripienaar** [comment]
ping @snoir please squash these and we can get it merged

**Comment by snoir** [comment]
@ripienaar is the commit ok ?

**Comment by ripienaar** [comment]
thank you @snoir

**Pull request merged**